### PR TITLE
Update/16128 add attribute types documentation

### LINKED ISSUE #16128: Add attribute types to docs
When registering a block you can define the attributes and their types, but I can't find any docs detailing what the allowed types are.

https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/block-api/block-attributes.md

I can guess what some of them are, but what about boolean? Is it boolean or bool? And number/ int/ float/ all of the above?

I think a proper list would be very helpful.

Thanks

### PR DESCRIPTION
**Pull request by AdamDTak2112**
## Description

Fixes #16128 . Added accepted attribute types for the Blocks API, including a link to the Wordpress REST API documentation for clarification. 

## How has this been tested?

Ran changes in docs through Visual Studio Code's markdown preview testing.

Changes should not affect any other code, as it was purely an addition to documentation. No existing documentation was altered.

## Types of changes

Only a change in documentation, as clarification was requested through #16128 .

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

**Review by Soean — APPROVED** [review]
Thanks for contributing.

File: docs/designers-developers/developers/block-api/block-attributes.md [hunk 1]
```diff
@@ -32,6 +32,20 @@ _Example_: Extract the `src` attribute from an image found in the block's markup
...
+* null
+* boolean
+* object
+* array
+* number
+* string
+* integer
+	- Matches any number with a zero fractional part
+
+Additional details are found in the documentation for Wordpress's REST API [here](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/).
```
Comment (mkaz) [inline docs/designers-developers/developers/block-api/block-attributes.md]: Here's a suggestion for improving the link:

```suggestion
See [Wordpress's REST API documentation](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/) for additional details.
```

**Review by mkaz — APPROVED** [review]
Looks good, I added one suggestion for the link.

Also, not a big deal, but you have a definition for integer, but not number. I would add for both or neither, just the types seems sufficient to me.

File: docs/designers-developers/developers/block-api/block-attributes.md (hunk 1 above)
Comment (chrisvanpatten) [reply docs/designers-developers/developers/block-api/block-attributes.md]: Also, capital P :)

```suggestion
See [WordPress's REST API documentation](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/) for additional details.
```

Comment (mkaz) [reply docs/designers-developers/developers/block-api/block-attributes.md]: hah, how'd I miss that? :-)

**Review by chrisvanpatten — CHANGES_REQUESTED** [review]
Just adding a block on the PR until this change (a combo of mine + @mkaz's changes) is merged in, then I'd love to see this landed. Thanks!

```suggestion
See [WordPress's REST API documentation](https://developer.wordpress.org/rest-api/extending-the-rest-api/schema/) for additional details.
```

File: docs/designers-developers/developers/block-api/block-attributes.md [hunk 2]
```diff
@@ -32,6 +32,20 @@ _Example_: Extract the `src` attribute from an image found in the block's markup
 }
 // { "url": "https://lorempixel.com/1200/800/" }
 ```
+#### `attribute` Type Validation
```
Comment (chrisvanpatten) [inline docs/designers-developers/developers/block-api/block-attributes.md]:
```suggestion

#### `attribute` Type Validation
```

**Comment by AdamDTak2112** [comment]
I've taken the suggestions from @mkaz and @chrisvanpatten. I committed the changes, and please let me know if there is anything else that needs to be change.

Thank you so much for letting me contribute to this! I promise I'll capitalize the P from now on :)

**Review by chrisvanpatten — APPROVED** [review]

**Comment by Soean** [comment]
Thanks for the changes and welcome as a new Gutenberg contributor!

**Pull request merged**